### PR TITLE
Macios symbolicate

### PR DESCRIPTION
**Pull request by jstedfast**
Spec: https://docs.google.com/document/d/1dvJVMhG3naxm247XbNE-2IhbF88K-N7ogfDa1xzXUBU/edit#heading=h.yi6nd7uot6wj

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by rolfbjarne** [comment]
This is unconditionally enable mSYMs by default, however iirc it was decided to only enable it by default for release builds (because it's quite useless for debug builds, and still adds to build time).

**Comment by jstedfast** [comment]
Something has to enable it or disable it via the .csproj settings

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by rolfbjarne** [comment]
Isn't it now disabled by default instead of enabled by default? Can't you use `MtouchDebug` to choose a default?

**Comment by jstedfast** [comment]
I suppose, but the IDE is what overrides the MonoSymbolArchive variable in the .csproj.

If we want it strictly tied to MtouchDebug, then it probably doesn't make sense to have the MonoSymbolArchive variable or the --msym arg, right?

**Comment by rolfbjarne** [comment]
So the IDEs will write a `MonoSymbolArchive` variable to every csproj then?

We don't want it strictly tied to `MtouchDebug`, we want it _default_ to `!MtouchDebug`, unless otherwise specified.

**Comment by jstedfast** [comment]
> So the IDEs will write a MonoSymbolArchive variable to every csproj then?

Correct.

> We don't want it strictly tied to MtouchDebug, we want it default to !MtouchDebug, unless otherwise specified.

Okay, we can probably do that.

**Comment by jstedfast** [comment]
Slight problem: Xamarin.Mac does not have anything equivalent to MtouchDebug, so it would not be consistent if we use that for Xamarin.iOS.

**Comment by rolfbjarne** [comment]
It would be the variable which causes `/debug` to be passed to mmp, which seems to be `$(DebugSymbols)`: https://github.com/xamarin/xamarin-macios/blob/master/msbuild/Xamarin.Mac.Tasks/Xamarin.Mac.Common.targets#L469

**Comment by jstedfast** [comment]
Well, $(DebugSymbols) is not XM specific. It's the Csc compiler tool value which means it won't really be the same behavior as iOS...

**Comment by xamarin-release-manager** [comment]
Build success

File: msbuild/Xamarin.Mac.Tasks/Tasks/Mmp.cs [hunk 1]
```diff
@@ -76,12 +77,13 @@ protected override string GenerateFullPathToTool ()
 		protected override bool ValidateParameters ()
 		{
 			XamMacArch arch;
-			return Enum.TryParse<XamMacArch> (Architecture, true, out arch);
+			return Enum.TryParse (Architecture, true, out arch);
```
Comment (rolfbjarne) [inline msbuild/Xamarin.Mac.Tasks/Tasks/Mmp.cs]: This doesn't look related to the msym changes?

**Comment by rolfbjarne** [comment]
@jstedfast Could you rebase on master? There seems to be conflicts now.

**Comment by jstedfast** [comment]
rebased on master

**Comment by xamarin-release-manager** [comment]
Build success
